everestrs/everestrs/src/lib.rs

```diff
@@ -287,3 +340,86 @@ impl Runtime {
         (self.cpp_module).as_ref().unwrap().signal_ready(self);
     }
 }
+
+/// A store for our config values. The type is closely related to
+/// [ffi::ConfigField] and [ffi::ConfigType].
+#[derive(Debug)]
+pub enum Config {
+    Boolean(bool),
+    String(String),
+    Number(f64),
+    Integer(i64),
+}
+
+impl TryFrom<&Config> for bool {
+    type Error = ();
+    fn try_from(value: &Config) -> std::result::Result<Self, Self::Error> {
+        match value {
+            Config::Boolean(value) => Ok(*value),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&Config> for String {
+    type Error = ();
+    fn try_from(value: &Config) -> std::result::Result<Self, Self::Error> {
+        match value {
+            Config::String(value) => Ok(value.clone()),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&Config> for f64 {
+    type Error = ();
+    fn try_from(value: &Config) -> std::result::Result<Self, Self::Error> {
+        match value {
+            Config::Number(value) => Ok(*value),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&Config> for i64 {
+    type Error = ();
+    fn try_from(value: &Config) -> std::result::Result<Self, Self::Error> {
+        match value {
+            Config::Integer(value) => Ok(*value),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Interface for fetching the configurations through the C++ runtime.
+pub fn get_module_configs() -> HashMap<String, HashMap<String, Config>> {
```

Comment:
return the Json blob and deseriailze it like we do with everything else. It is not worth to have two different ways of handling deserialization in this module. And as mentioned, the code is in the codegen branch already.

Reply:
retracted.